assemblyline_service_utilities/testing/helper.py

```diff
@@ -503,36 +502,76 @@ def _data_compare(ih: IssueHelper, original, new, data_type, ignore_new_extra_fi
 
     @staticmethod
     def _file_compare(ih: IssueHelper, f_type, original, new):
-        oh_map = {x["sha256"]: x["name"] for x in original}
-        on_map = {x["name"]: x["sha256"] for x in original}
-        nh_map = {x["sha256"]: x["name"] for x in new}
-        nn_map = {x["name"]: x["sha256"] for x in new}
-
-        for sha256, name in oh_map.items():
-            if sha256 not in nh_map:
-                if name not in nn_map:
-                    ih.add_issue(f_type, ih.ACTION_MISSING, f"File '{name} [{sha256}]' missing from the file list.")
-                    continue
-
-                if sha256 != nn_map[name]:
-                    ih.add_issue(
-                        f_type,
-                        ih.ACTION_CHANGED,
-                        f"The sha256 of the file '{name}' has changed. {sha256} -> {nn_map[name]}",
-                    )
-                    continue
+        if original == new:
+            # These file lists are exactly identical, no need to process for differences
+            return
+        elif len(new) == len(original) and all(n in original for n in new):
+            # The file lists, although the contents might be out of order, are the same
+            return
+
+        # Prune out items in the lists where nothing changed
+        for file in list(original):
+            if file in original and file in new:
```

Comment:
No need to check that `file in original`. Because `file` is iterating over a copy `original` it must be. There is no chance that it is removed from `original` previously since `.remove` only removes the first equal item, so if there are two copies of the same `file` the second will still be in `original`.